src/Control/Distributed/Process/Node.hs

```diff
@@ -319,11 +322,38 @@ startServiceProcesses node = do
 
 -- | Force-close a local node
 --
--- TODO: for now we just close the associated endpoint
+-- This function first try to brutally kill all processes, this
+-- process is not guaranteed to continue. So this is application
+-- responsibility to either close all it's processes before dying
+-- or being able to gracefully shutdown.
 closeLocalNode :: LocalNode -> IO ()
-closeLocalNode node =
-  -- TODO: close all our processes, surely!?
-  NT.closeEndPoint (localEndPoint node)
+closeLocalNode node = join $ terminateProcesses
+  where
+    terminateProcesses :: IO (IO ())
```

Comment:
About why this solution was chosen. Currently we have following solutions:

1. original - same as this one but without looping at all. Basically this means that if any process will be spawned after start of `terminationProcess` they will survive. This solution may block any of process will refuse to die and may exit even if there are existing processes with a very large time frame for premature exit

2. my first attempt - had a loop but had no monitoring notifications. This solution had a tiny frame for prepature exit, but could loop if this process is in race with self restarting process. I.e. process that traps kill and spawn new process.

3. my second attempt (current one) - this is the same as 1 but with looping. Such looping reduces time frame for premature exit (and basically could entirely remote it), but can potentially block if there is self-restarting process.

Summing all this up. I think that we need to either decide to implement some heavyweight solution, or 3 could be the best fit.